### PR TITLE
fix(drizzle): update schema type & fix issue with default id

### PR DESCRIPTION
**Pull request by realmikesolo**
Made tables for adapter optional cause not all of them are used for authentication strategies. Also I exlcuded `id` from data coming from Authjs in `createUser` function to enable setting default id as defined in Drizzle table.

## ☕️ Reasoning

1. Default id was overwritten by one coming from Authjs;
2. Not all tables are used for `next-auth`.

## 🧢 Checklist

- [X] Documentation
- [X] Tests
- [ ] Ready to be merged

### CODE REVIEW

**Comment by srigi** [comment]
+1 on this, as I'm also not using INT or UUID for PKs in my tables and this would unblock the whole project moving forward.

File: packages/adapter-drizzle/src/lib/pg.ts
Comment (juliusmarminge) [inline packages/adapter-drizzle/src/lib/pg.ts:114]: could save some cycles and re-use the id coming from auth.js as the fallback?
```suggestion
          hasDefaultId ? insertData : { ...insertData, id }
```

Comment (ndom91) [reply packages/adapter-drizzle/src/lib/pg.ts:114]: True, the `id` coming from auth.js will already be the result of a `crypto.randomUUID()` call. Can you do this update in all db types (postgres, mysql, sqlite)? :pray:

Comment (realmikesolo) [reply packages/adapter-drizzle/src/lib/pg.ts:114]: yep, will update now

**Comment by ndom91** [comment]
Okay nice, looks better to me. Tests all pass as well. @juliusmarminge you want to take one last look over it before I merge it? :pray:

File: packages/adapter-drizzle/src/index.ts
Comment (juliusmarminge) [inline packages/adapter-drizzle/src/index.ts:102]: can use the new `AdapterAccountType` introduced in https://github.com/nextauthjs/next-auth/pull/10832 to avoid circular refernce issues

**Review by juliusmarminge — COMMENTED** [review]
Rest looks good to me :)

**Review by ndom91 — COMMENTED** [review]
One last thing, lets also update the example(s) in the actual doc page at `/docs/pages/getting-started/adapters/drizzle.mdx`.

Ideally we wouldn't have two copies of the example setup code. One in the JSDoc in the `index.ts` **and** this docs page, but we just migrated to these new docs recently. We'll probably cleanup the schema stuff from the jsdoc shortly.

**Comment by JipSterk** [comment]
Do we still want to use `AdapterAccount['type']` in the tests?

**Review by ndom91 — APPROVED** [review]

**Pull request merged**
